src/terminal.rs

```diff
@@ -27,37 +27,36 @@ fn rgb2ansi(r: u8, g: u8, b: u8) -> u8 {
 }
 
 pub fn as_terminal_escaped(
-    v: &[(highlighting::Style, &str)],
+    style: highlighting::Style,
+    text: &str,
     true_color: bool,
     colored: bool,
 ) -> String {
-    let mut s: String = String::new();
-    for &(ref style, text) in v.iter() {
-        let style = if !colored {
-            Style::default()
-        } else {
-            let color = if true_color {
-                RGB(style.foreground.r, style.foreground.g, style.foreground.b)
-            } else {
-                let ansi = rgb2ansi(style.foreground.r, style.foreground.g, style.foreground.b);
-                Fixed(ansi)
-            };
 
-            if style.font_style.contains(FontStyle::BOLD) {
-                color.bold()
-            } else if style.font_style.contains(FontStyle::UNDERLINE) {
-                color.underline()
-            } else if style.font_style.contains(FontStyle::ITALIC) {
-                color.italic()
-            } else {
-                color.normal()
-            }
+    let style = if !colored {
+        Style::default()
+    } else {
+        let color = if true_color {
+            RGB(style.foreground.r, style.foreground.g, style.foreground.b)
+        } else {
+            let ansi = rgb2ansi(style.foreground.r, style.foreground.g, style.foreground.b);
+            Fixed(ansi)
         };
 
-        write!(s, "{}", style.paint(text)).unwrap();
-    }
+        if style.font_style.contains(FontStyle::BOLD) {
+            color.bold()
+        } else if style.font_style.contains(FontStyle::UNDERLINE) {
+            color.underline()
+        } else if style.font_style.contains(FontStyle::ITALIC) {
+            color.italic()
+        } else {
+            color.normal()
+        }
+    };
 
-    s
+    let mut s: String = String::new();
+    write!(s, "{}", style.paint(text)).unwrap();
+    return s;
```

Comment:
you can remove all of this and just return `style.paint(text).to_string()`. also, the return statement is not necessary. the last expression in a function will be returned